easy_my_coop/models/coop.py

```diff
@@ -102,9 +99,9 @@ def create_comp_sub_req(self, vals):
                 vals.get("company_register_number")
             )
             if cooperator:
+                vals["type"] = "subscription"
+                vals = self.is_member(vals, cooperator)
                 vals["partner_id"] = cooperator.id
-                vals["type"] = "increase"
-                vals["already_cooperator"] = True
```

Comment:
`is_member` does not do what it says it does (check whether cooperator is a a member and return a boolean).
Plus we could be more explicit in what the values passed to vals are => simpler code.

I'd suggest:
```py
    def is_member(self, vals, cooperator):  # remove or rename, avoid changing the state of vals 
        if cooperator.member:
            vals["type"] = "increase"
            vals["already_cooperator"] = True
        return vals

    @api.model
    def create(self, vals):

...
            if cooperator:
                if cooperator.is_member:
                    cooperator_values = {
                        "type": "increase",
                        "partner_id": cooperator.id,
                        "already_cooperator": True,
                    }
                else:
                    cooperator_values = {
                        "type": "subscription",
                        "partner_id": cooperator.id,
                        "already_cooperator": False,
                    }
                vals.update(cooperator_values)

```